### DISCUSSION TITLE
Support linear binning in `histogram`

### DISCUSSION THREAD

**Discussion by domoritz** [duckdb/duckdb#12900] [Ideas]
One major problem with binning is that if many values are near a threshold value, the histogram can look vastly different depending on how the bins are aligned. This makes the histograms less stable and downstream applications like smoothing may have to choose to use more bins to compensate for this issue. 

A simple solution is to use linear binning. The idea is that a value gets counted towards the two closest bins in the proportion of how close it is to the center of the bin. [Here](https://observablehq.com/@d3/contourdensity-linear-binning) is a nice demonstration of linear binning in two dimensions but you can see the same works in 1D. @jheer used linear binning in https://idl.uw.edu/papers/fast-kde as well. It would be nice to have linear binning in DuckDB as well. 

cc @Mytherin

**Comment by jheer** [comment]
I'll just add that [Mosaic vgplot](https://idl.uw.edu/mosaic/vgplot/) includes linear binning as part of both 1D and 2D kernel density estimation, using SQL union queries:
- https://github.com/uwdata/mosaic/blob/main/packages/plot/src/marks/Density1DMark.js#L91
- https://github.com/uwdata/mosaic/blob/main/packages/plot/src/marks/Grid2DMark.js#L241
